collector.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{BTreeSet, HashMap};
+use std::sync::{Arc, Mutex};
+
+use store_api::logstore::provider::KafkaProvider;
+use store_api::logstore::EntryId;
+use store_api::storage::RegionId;
+
+/// The [`IndexCollector`] trait defines the operations for managing and collecting index entries.
+pub trait IndexCollector: Send + Sync {
+    /// Sets the latest [`EntryId`].
+    fn set_latest_entry_id(&mut self, entry_id: EntryId);
+
+    /// Appends an [`EntryId`] for a specific region.
+    fn append(&mut self, region_id: RegionId, entry_id: EntryId);
+
+    /// Truncates the index for a specific region up to a given [`EntryId`].
```

Comment:
Is the goal of truncate to include `entry_id`? Or does it only include all entries smaller than `entry_id`? Please clarify in the comments.